#! /usr/bin/env python
# Task Long configuration

PACKAGE='sim_tasks'
import roslib; roslib.load_manifest(PACKAGE)

from dynamic_reconfigure.parameter_generator import *
from task_manager_lib.parameter_generator import *

gen = TaskParameterGenerator()
#       Name                    Type            Description
#       Default    Min    Max
gen.add("angle",          double_t, 0,    "Goal angle between the boat and the shore",  3.14)
gen.add("distance",       double_t, 0,    "Reference distance to reached between the boat and the other shore",  3.0)
gen.add("rotation_vel",   double_t, 0,    "Absolute angular velocity of the boat",  0.4)
gen.add("velocity",       double_t, 0,    "Linear velocity of the boat",  0.8)
gen.add("angle_range",    double_t, 0,    "Range on each side of the reference angle, where we detect the points",  0.78)
gen.add("angle_error_max",double_t, 0,    "Saturation of the angle error",  0.78)
gen.add("k_scal_prod",    double_t, 0,    "Gain for lateral distance control",  0.2)
gen.add("k_alpha",        double_t, 0,    "Gain for angular control",  0.2)

exit(gen.generate(PACKAGE, "sim_tasks", "TaskJoinOtherShore"))

